components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -2079,6 +2081,57 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, AutoContribution) {
   rewards_service_->RemoveObserver(this);
 }
 
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, AutoContributeWhenACOff) {
+    // Observe the Rewards service
+  rewards_service_->AddObserver(this);
+
+  EnableRewards();
+
+  // Claim grant using panel
+  const bool use_panel = true;
+  ClaimGrant(use_panel);
+
+  // Visit verified publisher
+  const bool verified = true;
+  VisitPublisher("duckduckgo.com", verified);
+
+  // toggle auto contribute off
+  content::EvalJsResult toggleOffResult = EvalJs(contents(),
+    "let toggleClicked = false;"
+    "new Promise((resolve) => {"
+    "var count = 10;"
+    "var interval = setInterval(function() {"
+    "  if (count == 0) {"
+    "    clearInterval(interval);"
+    "    resolve(false);"
+    "  } else {"
+    "    count -= 1;"
+    "  }"
+    "  if (document.querySelector(\"[data-test-id2='autoContribution']\")) {"
+    "    if (!toggleClicked) {"
+    "      toggleClicked = true;"
+    "      document.querySelector("
+    "          \"[data-test-id2='autoContribution']\").click();"
+    "    } else {"
+    "      clearInterval(interval);"
+    "      resolve(document.querySelector("
+    "          \"[data-test-id2='autoContribution']\")"
+    "        .getAttribute(\"data-toggled\") === 'false');"
+    "    }"
+    "  }"
+    "}, 500);});",
```

Comment:
There's a `WaitForSelector` function that will handle some of this work for you.